### PR TITLE
Cortex LLM commands implementation

### PR DESCRIPTION
**Pull request by sfc-gh-pjob**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
* Added `snow cortex` commands:
    * `complete` - Given a prompt, the command generates a response using your choice of language model.
      In the simplest use case, the prompt is a single string. You may also provide a JSON file with conversation history including multiple prompts and responses for interactive chat-style usage.
    * `extract-answer` - Extracts an answer to a given question from a text document.
      The document may be a plain-English document or a string representation of a semi-structured (JSON) data object.
    * `sentiment` - Returns sentiment as a score between -1 to 1 (with -1 being the most negative and 1 the most positive, with values around 0 neutral) for the given English-language input text.
    * `summarize` - Summarizes the given English-language input text.
    * `translate` - Translates text from the indicated or detected source language to a target language.

### CODE REVIEW

File: src/snowflake/cli/plugins/cortex/commands.py [hunk 1]
```diff
@@ -1,6 +1,257 @@
...
+        show_default=False,
+    ),
+    model: Optional[str] = typer.Option(
+        DEFAULT_MODEL,
+        "--model",
+        help="String specifying the model to be used.",
+    ),
+    file: Optional[Path] = readable_file_option(
+        param_name="--file",
+        help_str="File containing conversation history to be used to generate a completion. Cannot be combined with TEXT argument.",
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/cortex/commands.py:45]: Should we point out this should be a json file?

File: src/snowflake/cli/plugins/cortex/commands.py [hunk 2]
```diff
@@ -1,6 +1,257 @@
...
+        show_default=False,
+    ),
+    from_language: Optional[str] = typer.Option(
+        None,
+        "--from",
+        help="String specifying the language code for the language the text is currently in. See Snowflake Cortex documentation for a list of supported language codes.",
+        show_default=False,
+    ),
+    to_language: str = typer.Option(
+        ...,
```
Comment (sfc-gh-turbaszek) [inline src/snowflake/cli/plugins/cortex/commands.py:220]: Should we use `en` as default?

File: tests/__snapshots__/test_help_messages.ambr [hunk 3]
```diff
@@ -978,6 +979,418 @@
...
+  │ --silent                         Turns off intermediate output to console.   │
+  ╰──────────────────────────────────────────────────────────────────────────────╯
+  
+  
+  '''
+# ---
+# name: test_help_messages[cortex.extract-answer]
+  '''
+                                                                                  
+   Usage: default cortex extract-answer [OPTIONS] [QUESTION]
```
[the remark this answers is not in the document]
Comment (sfc-gh-turbaszek) [reply tests/__snapshots__/test_help_messages.ambr:1064]: Sounds good to me 👌

File: src/snowflake/cli/plugins/cortex/manager.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+log = logging.getLogger(__name__)
+
+
+class CortexManager(SqlExecutionMixin):
+    def complete_for_prompt(
+        self,
+        text: Text,
+        model: Model,
+    ) -> CortexResultText:
```
[the remark this answers is not in the document]
Comment (sfc-gh-pjob) [reply src/snowflake/cli/plugins/cortex/manager.py]: Changed to `str`

File: src/snowflake/cli/plugins/cortex/models.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+from typing import NewType
```
[the remark this answers is not in the document]
Comment (sfc-gh-pjob) [reply src/snowflake/cli/plugins/cortex/models.py]: Changed to `types`

File: src/snowflake/cli/plugins/cortex/commands.py (hunk 1 above)
Comment (sfc-gh-pjob) [reply src/snowflake/cli/plugins/cortex/commands.py]: Yes, we should, changed.

File: src/snowflake/cli/plugins/cortex/commands.py (hunk 2 above)
Comment (sfc-gh-pjob) [reply src/snowflake/cli/plugins/cortex/commands.py:220]: I don't think so. The doc with proposal didn't present any default for target language, same as TRANSLATE function doesn't provide any default (https://docs.snowflake.com/en/sql-reference/functions/translate-snowflake-cortex)

Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/plugins/cortex/commands.py:220]: But most of the users are english-speaking. Not having to specify `--to=en` every time sounds like a good thing to have.

Comment (sfc-gh-pjob) [reply src/snowflake/cli/plugins/cortex/commands.py:220]: Discussed and agreed to keep it without a default value. Removing a default value in the future will be backward incompatible while adding it if needed will a simple compatible change.

**Pull request merged**